sky/packages/sky/lib/src/rendering/proxy_box.dart

```diff
@@ -126,6 +142,14 @@ class RenderConstrainedBox extends RenderProxyBox {
   String debugDescribeSettings(String prefix) => '${super.debugDescribeSettings(prefix)}${prefix}additionalConstraints: ${additionalConstraints}\n';
 }
 
+/// Forces child to layout at a specific aspect ratio
+///
+/// The width of this render object is the largest width permited by the layout
+/// constraints. The height of the render object is determined by applying the
+/// given aspect ratio to the width.
+///
+/// The aspect ratio is expressed as a ratio of width to height. For example,
+/// a 16:9 width:height aspect ratio would have a value of 16.0/9.0.
```

Comment:
Gotcha.  Alternatively, we could consider this a bug and use the same algorithm we use for images.